hooks/open_with_shotgun_launchapp.py

```diff
@@ -43,6 +43,13 @@ def execute(self, published_file, **kwargs):
             context = self.tank.context_from_entity("Task", published_file["task"].get("id"))
         else:
             context = self.tank.context_from_path(path)
+            # In case the path is not relative to the project, or the project has no schema,
+            # try to still get a relevant context from the entity or the project.
+            if not context or not context.project:
```

Comment:
Not sure if this is the right approach or may be the comment above is misleading: your code tries to get a fallback context, whatever made `self.tank.context_from_path` fail. Not sure why you're testing `context.project`, but I would say that if you get a context, you should simply use it. I'm assuming that `context_from_path` does not raise an exception but returns `None`?

Reply:
nope, the problem is that `context_from_path` does return a context, but it has only the url in it.

Reply:
:(. Please add a comment mentioning this and explaining that this the reason why we check `context.project`.